ui/src/app/job-list/job-list.component.html

```diff
@@ -2,7 +2,12 @@
   [showControls]="true"
   [jobs]="jobs"
   [pageSize]="pageSize"></jm-header>
-<jm-job-list-table
-  [dataSource]="dataSource"
-  (onJobsChanged)="handleJobsChanged()">
-</jm-job-list-table>
+<div class="table-container">
```

Comment:
Ideally, we'd have a directive here that changes the opacity of the table component. That would natively work with any theme and avoid the potential for bugs like mispositioned overlays. Material components can be a pain when it comes to styling, but have you tried this here?

Reply:
Could you expand on why you think that approach is ideal? Seems a bit unnatural to me, but I can see advantages e.g. this would look wrong if we changed the app's background color.

It also seems reasonable to block clicks to the table rows, which the overlay currently achieves, though this is not strictly necessary.

Reply:
In the past, I've run into bugs implementing overlays like this in Android. Stuff like: underlying content is clickable if you press exactly at the edges of the overlay, overlay appears misaligned in certain environments, content appears to flash on load in/out and might be briefly clickable depending on the order in which the framework constructs/tears down views. In the end I just replaced all the overlays with my aforementioned solution. I have no reason to believe that we'll see any of these bugs in particular, but opting for fewer moving parts exposes us to less HTML/CSS trickery. If you'd like the table to be unclickable while loading, directly disabling pointer actions on the table is also going to be safer than blocking access with an overlay to absorb mouse events.

This is more of a preemptive suggestion than a blocking change, so we can leave it as is unless it does actually cause problems.

Reply:
Done, kind of.

I moved the opacity setting onto the table element; lets try that out, I like the idea of not applying a particular color on the overlay.

Secondly though, I didn't remove the overlay. My primary purpose for that is to control the positioning of the spinner element (horizontally centered). The continues to have a secondary role of blocking clicks, which I don't feel particularly strongly about.